src/CoreBundle/Service/ActivePageService.php

```diff
@@ -252,8 +258,11 @@ private function modifyRouteParameters(array &$parameters, TdbCmsLanguage $langu
             if ($activePage->fieldPrimaryTreeIdHidden === $activePortal->fieldHomeNodeId) {
                 unset($parameters['pagePath']);
             } else {
-                $allPageRoutes = $this->routingUtil->getAllPageRoutes($activePortal, $language);
-                $parameters['pagePath'] = rtrim($allPageRoutes[$activePage->id]->getPrimaryPath(), '/');
+                $pagePath = $this->pageServiceUtil->getPagePath($activePage, $language);
+                if (true === $activePortal->fieldUseSlashInSeoUrls) {
```

Comment:
Shouldn't do PageServicUtil this?

Reply:
In theory yes, but I think PageServiceUtil needs to work exactly as is (trailing slashes in URLs are an astonishingly deep lava lake).